### PR TITLE
[Feature][linkis-engineplugin-spark] add a new Executor (json ETL)

### PR DESCRIPTION
**Pull request by rarexixi**
Add a new simple ETL feature，implemented by executing our configured json.

### Related issues/PRs

Related issues: #3435

### Brief change log

- Add a new simple ETL feature，implemented by executing our configured json.

### CODE REVIEW

File: linkis-engineconn-plugins/spark/pom.xml [hunk 1]
```diff
@@ -26,7 +26,7 @@
 
   <artifactId>linkis-engineplugin-spark</artifactId>
   <properties>
-    <spark.version>2.4.3</spark.version>
+    <spark.version>2.4.7</spark.version>
```
Comment (peacewong) [inline linkis-engineconn-plugins/spark/pom.xml:29]: There are many version upgrades and modifications, it is recommended to keep 2.4.3 for the time being

File: linkis-engineconn-plugins/spark/src/main/java/org/apache/linkis/engineplugin/spark/datacalc/exception/ConfigRuntimeException.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.linkis.engineplugin.spark.datacalc.exception;
+
+public class ConfigRuntimeException extends RuntimeException {
```
Comment (peacewong) [inline linkis-engineconn-plugins/spark/src/main/java/org/apache/linkis/engineplugin/spark/datacalc/exception/ConfigRuntimeException.java:20]: should extend LinkisRuntimeException

File: linkis-engineconn-plugins/spark/src/main/java/org/apache/linkis/engineplugin/spark/datacalc/exception/DatabaseNotConfigException.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.linkis.engineplugin.spark.datacalc.exception;
+
+public class DatabaseNotConfigException extends RuntimeException {
```
Comment (peacewong) [inline linkis-engineconn-plugins/spark/src/main/java/org/apache/linkis/engineplugin/spark/datacalc/exception/DatabaseNotConfigException.java:20]: should extend LinkisRuntimeException

File: linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/executor/SparkDataCalcExecutor.scala [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+import org.apache.orc.storage.common.io.Allocator.AllocatorOutOfMemoryException
+
+import java.lang.reflect.InvocationTargetException
+import java.util
+import java.util.concurrent.atomic.AtomicLong
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable.ArrayBuffer
+
+class SparkDataCalcExecutor(sparkEngineSession: SparkEngineSession, id: Long)
```
Comment (peacewong) [inline linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/executor/SparkDataCalcExecutor.scala:59]: Is it possible to inherit: SparkEngineConnExecutor

File: linkis-computation-governance/linkis-manager/linkis-label-common/src/main/java/org/apache/linkis/manager/label/entity/engine/RunType.scala [hunk 5]
```diff
@@ -43,4 +43,6 @@ object RunType extends Enumeration {
 
   val TRINO_SQL = Value("tsql")
 
+  val DATA_CALC = Value("data_calc") // spark datacalc (ETL)
```
[the remark this answers is not in the document]
Comment (rarexixi) [reply linkis-computation-governance/linkis-manager/linkis-label-common/src/main/java/org/apache/linkis/manager/label/entity/engine/RunType.scala:46]: Data calculation

File: linkis-engineconn-plugins/spark/src/main/java/org/apache/linkis/engineplugin/spark/datacalc/exception/ConfigRuntimeException.java (hunk 2 above)
Comment (rarexixi) [reply linkis-engineconn-plugins/spark/src/main/java/org/apache/linkis/engineplugin/spark/datacalc/exception/ConfigRuntimeException.java:20]: OK

File: linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/factory/SparkDataCalcExecutorFactory.scala [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+import org.apache.linkis.engineconn.computation.executor.execute.ComputationExecutor
+import org.apache.linkis.engineplugin.spark.entity.SparkEngineSession
+import org.apache.linkis.engineplugin.spark.exception.NotSupportSparkDataCalcTypeException
+import org.apache.linkis.engineplugin.spark.executor.SparkDataCalcExecutor
+import org.apache.linkis.manager.label.entity.Label
+import org.apache.linkis.manager.label.entity.engine.RunType
+import org.apache.linkis.manager.label.entity.engine.RunType.RunType
+
+/**
+ * xishihao_modify 添加 data_calc 类型
```
Comment (peacewong) [inline linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/factory/SparkDataCalcExecutorFactory.scala:32]: need remove

**Review by peacewong — DISMISSED** [review]
LGTM.

**Review by peacewong — APPROVED** [review]

**Pull request merged**